v2/protocol/kafka_sarama/receiver.go

```diff
@@ -114,9 +114,7 @@ func (c *Consumer) OpenInbound(ctx context.Context) error {
 
 	errCh := make(chan error)
 
-	go func(errs chan error) {
-		errs <- cg.Consume(context.Background(), []string{c.topic}, c)
-	}(errCh)
+	go c.startConsumerGroupLoop(cg, ctx, errCh)
 
 	select {
 	case <-ctx.Done():
```

Comment:
The select statement blocks until it gets an error from `errCh` or the context is done.
If `cg.Consume(...)` returns, for example, `ErrClosedClient`, we stop consuming and the select statement will block until it gets a signal from the context. 

Is that what we want?

Reply:
> Is that what we want?

yep, all `OpenInbound` implementations are intended to block: https://github.com/cloudevents/sdk-go/blob/master/v2/protocol/lifecycle.go#L9